addons/sale_project/models/project_milestone.py

```diff
@@ -7,14 +7,47 @@ class ProjectMilestone(models.Model):
     _name = 'project.milestone'
     _inherit = 'project.milestone'
 
+    def _default_sale_line_id(self):
+        project_id = self._context.get('default_project_id')
+        if project_id:
+            return self.env['project.project'].browse(project_id).sale_line_id.filtered(lambda sol: sol.qty_delivered_method == 'milestones')
+
     allow_billable = fields.Boolean(related='project_id.allow_billable')
     project_partner_id = fields.Many2one(related='project_id.partner_id')
 
-    sale_line_id = fields.Many2one('sale.order.line', 'Sales Order Item', help='Sales Order Item that will be updated once the milestone is reached.',
+    sale_line_id = fields.Many2one('sale.order.line', 'Sales Order Item', default=_default_sale_line_id, help='Sales Order Item that will be updated once the milestone is reached.',
         domain="[('order_partner_id', '=?', project_partner_id), ('qty_delivered_method', '=', 'milestones')]")
-    quantity_percentage = fields.Float('Quantity', help='Percentage of the ordered quantity that will automatically be delivered once the milestone is reached.')
+    sale_line_name = fields.Text(compute="_compute_sale_line_id_name")
+    quantity_percentage = fields.Float('Quantity Percentage', compute="_compute_quantity_percentage", store=True, help='Percentage of the ordered quantity that will automatically be delivered once the milestone is reached.')
+    product_uom_qty = fields.Float(related="sale_line_id.product_uom_qty")
+    product_uom = fields.Many2one(related="sale_line_id.product_uom")
+    total_qty = fields.Float("Total Quantity", compute="_compute_total_qty", inverse="_inverse_total_qty")
+    remaining_percentage = fields.Float("Remaining Quantity", compute="_compute_remaining_quantity")
+
+    @api.depends('sale_line_id.name')
+    def _compute_sale_line_id_name(self):
+        for rec in self:
+            rec.sale_line_name = rec.sale_line_id.display_name
```

Comment:
Why cant this field be "related"?

Reply:
If not, it rather depends on `sale_line_id.display_name` :)

Reply:
> Why cant this field be "related"?

I have updated the code as per your suggestion.thanks for review.

Reply:
Thank you to you :smile: